### DISCUSSION TITLE
Better calendar defaults for the historical statistics

### DISCUSSION THREAD

**Discussion by vermorel** [plausible/analytics#329] [Anything Else]
Most analytics apps (including Google Analytics) do a terrible job at reporting their historical statistics because they don't take into account the two dominant cyclical patterns that drive web traffic. It would be awesome if plausible where to provide those better default.

Web traffic follows (nearly always) two strong cyclical patterns: day of the week, and week of the year (seasonality). Not "abiding" by those cyclicities make the result hard to read and confusing. Thus I would suggest a short series of changes in the default of the calendar selection:

* Instead of the `Last 30 days`, I suggest the `Last 28 days`. Sticking to full weeks make numbers easier to interpret, precisely due to the day-of-the-week effect.
* I would suggest to remove the `Last 7 days`. Frankly, I am not to sure to see any practical situation where this report would be more useful than the last 28 days which is going to be fine even on mobile.
* I would suggest to remove `Last month` and `Previous month` by a single report `Current vs Previous Month` that superposes the last 28 days (yes, it's cheating) with the previous 28 days. 
* I would suggest to weekly aggregate the `Last 6 months` and `Last 12 months`. Cheating again, but the monthly statistics are just invariably skewed by the day-of-week effects. Display a timeline with 52 points is just fine. Also, the `Last 6 months` can probably be removed, I doubt that it serves any real purpose.

Finally, it would be really awesome if there was a `Current vs Previous Year` view where the last 52 weeks are compared to the 52 weeks that precede. Indeed, as web traffic tends to be highly seasonal, the only way to make sense of the traffic is compare to the relevant period last year.

**Comment by ukutaht** [comment]
Thanks for the suggestions @vermorel, this is very useful.

I agree with the first 2 points.

> I would suggest to remove Last month and Previous month by a single report Current vs Previous Month that superposes the last 28 days (yes, it's cheating) with the previous 28 days.

What would the difference be between this view and and the `Last 28 days` one? As far as I understand they would be identical no?

The reason main purpose of having `This month` and `Previous month` is that many businesses (ourselves included) report stats based on calendar months. So we wanted to have both rolling 30 day view as well as calendar-based months.

> I would suggest to weekly aggregate the Last 6 months and Last 12 months. Cheating again, but the monthly statistics are just invariably skewed by the day-of-week effects. Display a timeline with 52 points is just fine. Also, the Last 6 months can probably be removed, I doubt that it serves any real purpose.

Agreed that 6 months is pretty pointless if we have 12 months.

One idea we've had is to allow you to select both the timeframe (e.g. `Last 12 months`) and also the reporting interval (`monthly`, `weekly` or `daily`).

**Comment by vermorel** [comment]
The difference between 28 days and 30 days is minor, and clearly this isn't a showstopper. At Lokad (we specialize in statistics), I routinely witness a client or an employee misinterpreting the data in front of their eyes because they misread the day-of-week pattern. Thus, it would just a be a minor thing to help your userbase fall into the "pitfall of success".

About the monthly reports, I see you point. My professional opinion remains that monthly statistics are going to be accidentally noisy, but I understand that some businesses may want this, and it's fine by me. I would still suggest to unify those two reports into one, as the comparison would be more than the sum of the two independent reports.

Letting the end user select daily/weekl/monthly is the best. I did not suggest this because it's more dev efforts and I wanted to focus on the low hanging fruits, but clearly if you are willing to go down this path it would be awesome.
